CenterForOpenScience/osf.io
# [Feature] Fix fangorn edit undefined [OSF-7132]

## Purpose

Occasionally when moving great number of files (I tested between 10 and 70) between folders, one of the files variables will not be fully instantiated. When this occurs fangorn completely locks up and a page refresh is required.
## Changes

Ensure object property `permissions` exists before `permissions.edit` is checked.
## Side effects

Should be none based on where this check occurs. Each node is checked for permission to see if at least one can be deleted to decide if the delete button shows up. Because this bug requires a move action to occur, skipping one file that does not yet have its proper permission set will just hit the next, which (because it was moved) should have the correct permission.
## Ticket

https://openscience.atlassian.net/browse/OSF-7132
[#OSF-7132]
## QA Considerations
1. Try replicating the bug on production so you can see what happens when it fails. 
2. Test moving large numbers of files between folders in OSF storage. If it works after a number of moves, everything should be good. 
3. For added measure, make sure the delete button appears properly.
